### DISCUSSION TITLE
v0.4.0

### DISCUSSION THREAD

**Discussion by stefansundin** [stefansundin/privatkopiera#205] [Announcements]
- Tillägget har uppgraderats till Manifest Version 3.
  - Nya minimikrav på webbläsarversioner: Chrome 102 och Firefox 109.
  - Man behöver inte längre bevilja behörigheter för vissa hemsidor som förut.
  - Det är största omskrivningen av tillägget någonsin.
- Fixat inloggad tv4play.se.
- Uppdaterad detektering på svt.se.
  - Detta är tyvärr ganska bräckligt och kan nog sluta fungera igen om de uppdaterar sidan igen.
- Små förbättringar på urplay.se.
- Fixat buggar på dr.dk och nu stöds undertexter.
- Små förbättringar till utseendet (bootstrap uppdaterad till 5.3.2).
- Standardfiländelse har ändrats tillbaka till `.mp4` för att kompatibiliteten är bättre.
  - Ändra tillbaka till `.mkv` i inställningarna om du föredrar det istället.
- Nya inställningar:
  - FFmpeg-sökvägen kan specificeras. Du kan använda en absolut sökväg, t.ex. `C:\ffmpeg\ffmpeg.exe`. Du kan också lägga till nya argument till ffmpeg.
  - Du kan specificera sökvägen där dina filer sparas, t.ex. `C:\Användare\Svensson\Skrivbord\`.

**Full Changelog**: https://github.com/stefansundin/privatkopiera/compare/v0.3.10...v0.4.0

<hr /><em>This discussion was created from the release <a href='https://github.com/stefansundin/privatkopiera/releases/tag/v0.4.0'>v0.4.0</a>.</em>

**Comment by stefansundin** [comment]
Om någon märker av någon bugg så rapportera det gärna här.

**Reply by kronosjo** [reply]
Undertext för svtplay fungerar ej.
Sparar man bara textfilen så innehåller den inget. 0 kb.

**Reply by Lunarwalker60** [reply]
Jag upptäckte detsamma. Stefan; se upp med text-länken - ibland är den 'text/text-closed' och ibland är den 'text/text-open'. För mig funkade det när jag ändrade t '-open' för Moonage - om D Bowie.

**Reply by stefansundin** [reply]
@kronosjo @Lunarwalker60 Ni stöter nog på problemet som är beskrivet i #120. Testa FFmpeg från https://github.com/yt-dlp/FFmpeg-Builds#downloads.

**Reply by stefansundin** [reply]
Har uppdaterat hemsidan nu och rekommenderar numera att ladda ned FFmpeg härifrån.

**Comment by stefansundin** [comment]
Ett tips med den nya inställningen där man ställer in sökvägen där filerna sparas är att man kan använda systemvariabler.

- Windows: t.ex. `%USERPROFILE%\Skrivbord\`
  - https://www.tenforums.com/tutorials/121855-edit-user-system-environment-variables-windows.html
- macOS och Linux: t.ex. `$HOME/Downloads/`

**Comment by GunnarHallgren** [comment]
Verkar vara problem med några svt filmer  tex avlyssningen 
C:\Users\Gunna>ffmpeg -i "https://svt-vod-secure-2c.akamaized.net/d0/secure/20231221/6a10af58-bd2f-4fe6-98e3-5bf37ae81c10/hls-video-avc-1920x1080p25-3095/hls-video-avc-1920x1080p25-3095.m3u8?aka_me_session_id=AAAAAAAAAADVQpllAAAAAM2RLDr7tzjAZYIL3ENF4lECvC158da3UX%2fjL574SfweSM89dTgJ+Yj3IVF22xtPuiVh4dSNyYEY&aka_media_format_type=hls" -i "https://svt-vod-secure-2c.akamaized.net/d0/secure/20231221/6a10af58-bd2f-4fe6-98e3-5bf37ae81c10/hls-audio-sv-aac-2ch-192/hls-audio-sv-aac-2ch-192.m3u8?aka_me_session_id=AAAAAAAAAADVQpllAAAAAM2RLDr7tzjAZYIL3ENF4lECvC158da3UX%2fjL574SfweSM89dTgJ+Yj3IVF22xtPuiVh4dSNyYEY&aka_media_format_type=hls" -i "https://svt-vod-secure-2c.akamaized.net/d0/secure/20231221/6a10af58-bd2f-4fe6-98e3-5bf37ae81c10/text/text-open.vtt" -vcodec copy -acodec copy -absf aac_adtstoasc "Avlyssningen.mp4"

resulterar i 

hls @ 0000026fa494bd00] Opening 'crypto+https://svt-vod-secure-2c.akamaized.net/d0/secure/20231221/6a10af58-bd2f-4fe6-98e3-5bf37ae81c10/hls-video-avc-1920x1080p25-3095/hls-video-avc-1920x1080p25-3095-1.ts?aka_me_session_id=AAAAAAAAAADVQpllAAAAAM2RLDr7tzjAZYIL3ENF4lECvC158da3UX%2fjL574SfweSM89dTgJ+Yj3IVF22xtPuiVh4dSNyYEY&aka_msn=0&aka_hls_version=4&aka_media_format_type=hls' for reading
Input #0, hls, from 'https://svt-vod-secure-2c.akamaized.net/d0/secure/20231221/6a10af58-bd2f-4fe6-98e3-5bf37ae81c10/hls-video-avc-1920x1080p25-3095/hls-video-avc-1920x1080p25-3095.m3u8?aka_me_session_id=AAAAAAAAAADVQpllAAAAAM2RLDr7tzjAZYIL3ENF4lECvC158da3UX%2fjL574SfweSM89dTgJ+Yj3IVF22xtPuiVh4dSNyYEY&aka_media_format_type=hls':
  Duration: 01:49:02.08, start: 0.100000, bitrate: 0 kb/s
  Program 0
    Metadata:
      variant_bitrate : 0
  Stream #0:0: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p(tv, bt709/unknown/bt709), 1920x1080 [SAR 1:1 DAR 16:9], 25 fps, 25 tbr, 90k tbn
    Metadata:
      variant_bitrate : 0
[hls @ 0000026fa5c20d40] Skip ('## Generated with https://github.com/google/shaka-packager version v2.6.1-634af65-release')
[hls @ 0000026fa5c20d40] Skip ('## Post-processed by Team Video Core @ SVT')
[hls @ 0000026fa5c20d40] Skip ('#EXT-X-VERSION:4')
[hls @ 0000026fa5c20d40] Opening 'https://svt-vod-secure-2c.akamaized.net/d0/secure/20231221/6a10af58-bd2f-4fe6-98e3-5bf37ae81c10/hls-audio-sv-aac-2ch-192/serve.key?aka_me_session_id=AAAAAAAAAADVQpllAAAAAM2RLDr7tzjAZYIL3ENF4lECvC158da3UX%2fjL574SfweSM89dTgJ+Yj3IVF22xtPuiVh4dSNyYEY&aka_media_format_type=hls' for reading
[hls @ 0000026fa5c20d40] Opening 'crypto+https://svt-vod-secure-2c.akamaized.net/d0/secure/20231221/6a10af58-bd2f-4fe6-98e3-5bf37ae81c10/hls-audio-sv-aac-2ch-192/hls-audio-sv-aac-2ch-192-1.aac?aka_me_session_id=AAAAAAAAAADVQpllAAAAAM2RLDr7tzjAZYIL3ENF4lECvC158da3UX%2fjL574SfweSM89dTgJ+Yj3IVF22xtPuiVh4dSNyYEY&aka_msn=0&aka_hls_version=4&aka_media_format_type=hls' for reading
Input #1, hls, from 'https://svt-vod-secure-2c.akamaized.net/d0/secure/20231221/6a10af58-bd2f-4fe6-98e3-5bf37ae81c10/hls-audio-sv-aac-2ch-192/hls-audio-sv-aac-2ch-192.m3u8?aka_me_session_id=AAAAAAAAAADVQpllAAAAAM2RLDr7tzjAZYIL3ENF4lECvC158da3UX%2fjL574SfweSM89dTgJ+Yj3IVF22xtPuiVh4dSNyYEY&aka_media_format_type=hls':
  Duration: 01:49:02.12, start: 0.057333, bitrate: 0 kb/s
  Program 0
    Metadata:
      variant_bitrate : 0
  Stream #1:0: Audio: aac (LC), 48000 Hz, stereo, fltp, 194 kb/s
    Metadata:
      variant_bitrate : 0
      id3v2_priv.com.apple.streaming.transportStreamTimestamp: \x00\x00\x00\x00\x00\x00\x14(
[https @ 0000026fa53da640] HTTP error 403 Forbidden
https://svt-vod-secure-2c.akamaized.net/d0/secure/20231221/6a10af58-bd2f-4fe6-98e3-5bf37ae81c10/text/text-open.vtt: Server returned 403 Forbidden (access denied)

**Comment by spirit03** [comment]
Hej!
Nu när SVT stödjer 2k och 4k så undrar jag om privat kopiera stödjer det. Testade att ladda ned en titel som finns i 4k på svtplay med ffmpeg, men fick då inte mer än 1080p. Är det ffmpeg som inte stödjer högre upplösning än 1080p eller ligger begräsningen någon annanstans?

**Reply by stefansundin** [reply]
Fixat i v0.4.2.
